File: backend-nestjs/src/modules/auth/dto/email-verification-response.dto.ts
export class EmailVerificationResponseDto {
  success!: boolean;
  message!: string;
  expiresInMinutes?: number;

  static success(message: string, expiresInMinutes?: number): EmailVerificationResponseDto {
    return {
      success: true,
      message,
      expiresInMinutes,
    };
  }

  static error(message: string): EmailVerificationResponseDto {
    return {
      success: false,
      message,
    };
  }

  static builder() {
    return new EmailVerificationResponseBuilder();
  }
}

class EmailVerificationResponseBuilder {
  private dto: EmailVerificationResponseDto = { success: false, message: '' };

  success(value: boolean): this {
    this.dto.success = value;
    return this;
  }

  message(value: string): this {
    this.dto.message = value;
    return this;
  }

  expiresInMinutes(value: number): this {
    this.dto.expiresInMinutes = value;
    return this;
  }

  build(): EmailVerificationResponseDto {
    return this.dto;
  }
}

File: backend/src/main/java/com/northwollo/tourism/dto/response/EmailVerificationResponseDto.java
package com.northwollo.tourism.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EmailVerificationResponseDto {

    private String message;
    private boolean success;
    private Integer expiresInMinutes; // OTP expiry time

    public static EmailVerificationResponseDto success(String message) {
        return EmailVerificationResponseDto.builder()
                .message(message)
                .success(true)
                .build();
    }

    public static EmailVerificationResponseDto error(String message) {
        return EmailVerificationResponseDto.builder()
                .message(message)
                .success(false)
                .build();
    }
}